pyintegration/deephaven2/dtypes.py

```diff
@@ -146,3 +147,20 @@ def j_array_list(values: Iterable):
         return j_list
     except Exception as e:
         raise DHError(e, "failed to create a Java ArrayList from the Python collection.") from e
+
+
+def _get_full_type_name(obj):
+    instances = re.findall("<class '(.*?)'>", str(type(obj)))
+    if len(instances) > 0:
+        return instances[0]
+    else:
+        return None
+
+
+def is_java_type(obj, return_name=False):
+    class_name = _get_full_type_name(obj)
+    value = (class_name is not None) and (class_name in jpy.types)
```

Comment:
why not use `isinstance(obj, jpy.JType)`?

Reply:
And then, if we know it's a java type, we can do:

```
obj.jclass.getName()
```

Reply:
good idea, this is from the old conversion_utils.py, I probably shouldn't have just copied without questioning it.

Reply:
My recollection is that I tried this `isinstance(obj, jpy.JType)` and it did not work consistently.  Be sure to test thoroughly if you try this change.

Reply:
If we find any exceptional cases where we think something should be JType, but isn't, we need to capture that information in a ticket.

We do need some sort of ABC eventually (if we don't already) for all of our handwrapped python classes, such that we know they are handwrapped java types.

Come across that a few times, and I've made notes about it.

* https://github.com/deephaven/deephaven-plugin-json/blob/v0.0.1.dev3/src/deephaven/plugin/json/__init__.py#L49-L51
* https://github.com/devinrsmith/deephaven-core/blob/a7fc020d58401ba231f8fcf81f27d9b50fe9e43b/pyintegration/deephaven2/server/script_session/__init__.py#L32-L46